akka-remote/src/main/resources/reference.conf

```diff
@@ -1176,6 +1175,66 @@ akka {
+        # This provider also includes a verification fase that runs after the TLS handshake
+        # phase. In this verification, both peers run an authorization and verify they are
+        # part of the same akka cluster. The verification happens via comparing the subject
+        # names in the peer's certificate with the name on the own certificate so if you
+        # use this SSLEngineProvider you should make sure all nodes on the cluster include
+        # at least one common subject name (CN or SAN).
+        # The Key setup this implementation supports has some limitations:
+        #   1. the private key must be provided on a PKCS#1 or a non-encrypted PKCS#8 PEM-formatted file
+        #   2. the private key must be be of an algorythm supported by `akka-pki` tools (e.g. "RSA", not "EC")
+        #   3. the node certificate must be issued by a root CA (not an intermediate CA)
+        #   4. both the node and the CA certificates must be provided in PEM-formatted files
+        rotating-keys-engine {
+
+          # This is a convention that people may follow if they wish to save themselves some configuration
+          secret-mount-point = /var/run/secrets/akka-tls/rotating-keys-engine
+
+          # The absolute path the PEM file with the private key.
+          key-file = ${akka.remote.artery.ssl.rotating-keys-engine.secret-mount-point}/tls.key
+          # The absolute path to the PEM file of the certificate for the private key above.
+          cert-file = ${akka.remote.artery.ssl.rotating-keys-engine.secret-mount-point}/tls.crt
+          # The absolute path to the PEM file of the certificate of the CA that emited
+          # the node certificate above.
+          ca-cert-file = ${akka.remote.artery.ssl.rotating-keys-engine.secret-mount-point}/ca.crt
+
+          # There are two options, and the default SecureRandom is recommended:
+          # "" or "SecureRandom" => (default)
+          # "SHA1PRNG" => Can be slow because of blocking issues on Linux
+          #
+          # Setting a value here may require you to supply the appropriate cipher
+          # suite (see enabled-algorithms section)
+          random-number-generator = ${akka.remote.artery.ssl.config-ssl-engine.random-number-generator}
+
+          # Example: ["TLS_DHE_RSA_WITH_AES_128_GCM_SHA256",
+          #   "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+          #   "TLS_DHE_RSA_WITH_AES_256_GCM_SHA384",
+          #   "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384"]
+          # If you use a JDK 8 prior to 8u161 you need to install
+          # the JCE Unlimited Strength Jurisdiction Policy Files to use AES 256.
+          # More info here:
+          # https://www.oracle.com/java/technologies/javase-jce-all-downloads.html
+          enabled-algorithms = ["TLS_DHE_RSA_WITH_AES_128_GCM_SHA256",
+             "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+             "TLS_DHE_RSA_WITH_AES_256_GCM_SHA384",
+             "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384"]
```

Comment:
Order here matters, it specifies the preference. Since typically all nodes will have the same configuration, this means `TLS_DHE_RSA_WITH_AES_128_GCM_SHA256` will always be selected, and the other three will never be considered. Is that the best out of the 4? I would have suggested `TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384` as the top one, since `ECDHE` is considered more secure than `DHE` (there are known ways to approximate factoring primes that lead to weaknesses in `DHE`, and new ways that make it weaker are always coming up), whereas there's no equivalent known weaknesses in eliptic curve cryptography), and `SHA384` is faster on 64 bit hardware than `SHA256`.

I would actually recommend only having one algorithm here too - there's just no reason to have more than one since both the client and server will share the same configuration, unless you're switching defaults, and then in one release you would add the new default as the first option and the old as the second, and then in a subsequent release, you would drop the second.